app/repositories/LookupRepository.scala

```diff
@@ -27,34 +27,35 @@ import uk.gov.hmrc.mongo.play.json.PlayMongoRepository
 
 import javax.inject.{Inject, Singleton}
 import scala.concurrent.{ExecutionContext, Future}
+import scala.util.Try
 
 trait LookupRepository {
 
-  def save(nino: String, mtdId: String): Future[Boolean]
+  def save(reference: MtdIdCached): Future[Boolean]
 
-  def getMtdReference(nino: String): Future[Option[MtdIdReference]]
+  def getMtdReference(nino: String): Future[Option[MtdIdCached]]
 
 }
 
 @Singleton
 class LookupRepositoryImpl @Inject() (mongo: MongoComponent)(implicit ec: ExecutionContext)
-    extends PlayMongoRepository[MtdIdReference](
+    extends PlayMongoRepository[MtdIdCached](
       collectionName = "mtdIdLookup",
       mongoComponent = mongo,
-      domainFormat = MtdIdReference.format,
+      domainFormat = MtdIdCached.format,
       indexes = Seq(
         IndexModel(ascending("nino"), IndexOptions().name("mtd-nino").unique(true).background(true))
       ),
       replaceIndexes = false
     )
     with LookupRepository {
 
-  def save(nino: String, mtdId: String): Future[Boolean] =
-    collection.insertOne(MtdIdReference(nino, mtdId)).toFuture().map(result => result.wasAcknowledged())
+  def save(reference: MtdIdCached): Future[Boolean] =
+    Try { collection.insertOne(reference).toFuture().map(result => result.wasAcknowledged()) }.getOrElse(Future.successful(false))
 
   def removeAll(): Future[DeleteResult] = collection.deleteMany(new BasicDBObject()).toFuture()
 
-  def getMtdReference(nino: String): Future[Option[MtdIdReference]] =
-    collection.find(equal("nino", nino)).toFuture().map(_.headOption)
+  def getMtdReference(nino: String): Future[Option[MtdIdCached]] =
+    Try { collection.find(equal("nino", nino)).toFuture().map(_.headOption) }.getOrElse(Future.successful(None))
```

Comment:
As for save maybe simply log the error as follows. This is slightly more resilient than the current main branch which would fail the whole request if the cache is not available.

```scala
def getMtdReference(nino: String): Future[Option[MtdIdCached]] =
    collection
      .find(equal("nino", nino))
      .toFuture()
      .map(_.headOption)
      .recover { case e =>
        logger.warn("Error retrieving cached reference", e)
        None
      }
```